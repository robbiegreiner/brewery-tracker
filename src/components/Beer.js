import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Beer extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    const { currentBeer, getBeerByID } = this.props;
    if (Object.keys(currentBeer)) {
      console.log('hey');
      getBeerByID(this.props.match.params.beer_id);
    }
  }
  renderBeer(beer){
    if (beer){
      return (
        <div className='big-beer'>
          <h1>{beer.name}</h1>
          <h3>{beer.breweries ? beer.breweries[0].name : 'no brewery'}</h3>
          <h3>{beer.style ? beer.style.name : 'no style'}</h3>
          <h3>{beer.abv}</h3>
          <p>{beer.description}</p>
        </div>
      );
    } else {
      <h1>Beer!</h1>;
    }
  }


  render() {
    return (
      <div>
        {this.renderBeer(this.props.currentBeer)}
      </div>
    );
  }

}

Beer.propTypes = {
  currentBeer: PropTypes.object
};

export default Beer;
