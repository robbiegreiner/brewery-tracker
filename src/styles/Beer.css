.big-beer {
  margin-top: 0px;
  background-color: white;
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center; }

.big-beer h1 {
  margin-top: 150px;
  font-size: 50px; }

.big-beer p {
  width: 700px;
  text-align: center; }

.big-beer-logo {
  width: 400px; }

.big-beer-icon {
  width: 300px;
  margin-left: 50px;
  margin-bottom: 50px; }

.big-beer h3, .big-beer h2 {
  margin-bottom: 10px; }

.big-beer a {
  color: black; }
