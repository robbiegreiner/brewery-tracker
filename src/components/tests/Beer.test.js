import React from "react";
import Beer from '../Beer';
import { shallow } from "enzyme";

describe('Beer component', () => {
  const mockFn = jest.fn();
  let wrapper = shallow(<Beer />);


  it('should match snapshot', () => {
    // console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });


});
