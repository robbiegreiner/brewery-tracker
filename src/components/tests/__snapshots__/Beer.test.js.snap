// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beer component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Beer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
